src/MSBuild/TerminalLogger/TerminalLogger.cs

```diff
@@ -469,6 +482,31 @@ private void BuildFinished(object sender, BuildFinishedEventArgs e)
         _testEndTime = null;
     }
 
+    private void RenderBuildSummary()
+    {
+        if (!_projects.Any(p => p.Value.GetBuildErrorMessages().Any()))
+        {
+            // No errors to display.
+            return;
+        }
+
+        Terminal.WriteLine(ResourceUtilities.GetResourceString("BuildSummary"));
+
+        foreach (Project project in _projects.Values.Where(p => p.GetBuildErrorMessages().Any()))
+        {
+            string projectFileName = Path.GetFileNameWithoutExtension(project.File);
+            string? tfm = project.TargetFramework;
+            Terminal.WriteLine($"{Indentation}{projectFileName}{(tfm is null ? string.Empty : " ")}{AnsiCodes.Colorize(tfm, TerminalLogger.TargetFrameworkColor)}");
```

Comment:
Feels like we should abstract this out if we're going to do it in 2 places.

Reply:
I wasn't sure whether the final implementation, after resolving all open questions, would be the same, so I kept a separate implementation for now.

Reply:
Yes - let's start with keeping the formatting in both the mid-build and build summary areas the same as much as possible - so:

* project name
* tfm (if known)
* diagnostic count
* list of diagnostics

should be present in the summary. The main thing we should consider _not_ writing is the dll output.

Reply:
Dll output is displayed for succeeded builds, and they should not be a part of the summary, if I understood it correctly.